[Joy] Add `Sheet` doc (#32820)
mui/material-ui: docs/data/joy/components/sheet/sheet.md

---
product: joy-ui
title: React Sheet component
---

# Sheet

<p class="description">Sheet is a useful surface component that implements the global variant feature.</p>

{{"component": "modules/components/ComponentLinkHeader.js"}}

Use the `Sheet` component as a generic container for your components.

> You can play around with variants, color, and elevation to achieve the desired result.

Reviewer: ```suggestion
You can play around with variants and colors to achieve the desired result.
```

---
product: joy-ui
title: React Sheet component

> githubLabel: 'component: sheet'

Reviewer: ```suggestion
```

Reviewer: let's not create a new label for now.

> unstyled: /base/react-sheet/

Reviewer: ```suggestion
```

Reviewer: There is no unstyled version of sheet.

---

# Sheet

<p class="description">Sheet is a useful surface component that implements the global variant feature.</p>

{{"component": "modules/components/ComponentLinkHeader.js"}}

Use the `Sheet` component as a generic container for your components.
You can play around with variants, color, and elevation to achieve the desired result.

> {{"demo": "SheetUsage.js", "hideToolbar": true}}

Reviewer: ```suggestion
{{"demo": "SheetUsage.js", "hideToolbar": true, "bg": true}}
```

---
product: joy-ui
title: React Sheet component
githubLabel: 'component: sheet'
unstyled: /base/react-sheet/
---

# Sheet

> <p class="description">Sheet generates a background element that can be used for various purposes.</p>

Reviewer: ```suggestion
<p class="description">Sheet is a useful surface component that implements the global variant feature.</p>
```

{{"component": "modules/components/ComponentLinkHeader.js"}}

## Basic

Use the `Sheet` component as a generic container for your components.
You can play around with variants, color, and elevation to achieve the desired result.

{{"demo": "SheetUsage.js", "hideToolbar": true}}

> ## Variants
> 
> The `Sheet` comes with four global variants: `plain` (default), `outlined`, `soft` and `solid`.
> 
> {{"demo": "SheetVariants.js"}}
> 
> ## Colors
> 
> All theme palettes are available via the `color` prop which can be combined with the `variant` prop.
> 
> {{"demo": "SheetColors.js"}}
> 
> ## Elevations
> 
> The `Sheet` comes with three predefined elevation values: `sm`, `md`, and `lg`.
> 
> {{"demo": "SheetElevations.js"}}

Reviewer: ```suggestion
```

I think we don't need these sections anymore because of the playground.
